impl/maven-core/src/main/java/org/apache/maven/lifecycle/internal/MojoExecutor.java

```diff
@@ -306,7 +306,7 @@ private void doExecute(MavenSession session, MojoExecution mojoExecution, Depend
 
         ensureDependenciesAreResolved(mojoDescriptor, session, dependencyContext);
 
-        try (NoExceptionCloseable lock = getProjectLock(session, mojoDescriptor)) {
+        try (NoExceptionCloseable ignore = getProjectLock(session, mojoDescriptor)) {
```

Comment:
Yes, looks like a bug in PMD

Reply:
really? Show you IDE screen not point out the same:

Looks like a bug in maven to me.

<img width="1331" alt="image" src="https://github.com/user-attachments/assets/d5fc78b9-51e8-435b-97b1-b7724b2ccd62" />

Reply:
It is used implicitly by the "try with resource" constructor. The `lock` name is fine, as it is a meaningful name if a future version decides to use it explicitly.

Reply:
This code makes no sense then. Why declare a field not used?

Then simply remove the overhead not needed:

<img width="1148" alt="image" src="https://github.com/user-attachments/assets/ef738267-e474-4549-bd61-4083935146f4" />

Reply:
set dedicate this:
- https://github.com/apache/maven/pull/2384